File: 028/cg_028_easy.java
public class Solution {
    public int strStr(String haystack, String needle) {
        if (needle.isEmpty()) {
            return 0;
        }

        for (int i = 0; i < haystack.length(); i++) {
            if (i + needle.length() > haystack.length()) {
                break;
            }

            if (haystack.substring(i, i + needle.length()).compareTo(needle) == 0) {
                return i;
            }
        }

        return -1;
    }
}

File: 028/cg_028_easy.py
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        if len(needle) < 1:
            return 0

        for i in range(0, len(haystack)):
            if (i + len(needle)) <= len(haystack) and haystack[i:i + len(needle)] == needle:
                return i
        return -1
